packages/amplify-provider-awscloudformation/lib/push-resources.js

```diff
@@ -475,6 +485,52 @@ function updateIdPRolesInNestedStack(context, nestedStack, authResourceName) {
   Object.assign(nestedStack.Resources, idpUpdateRoleCfn);
 }
 
+function packageLayer(context, resource) {
```

Comment:
I would still recommend moving this to the function category plugin and importing it here. Logic specific to category functionality doesn't belong in the provider plugin